docusaurus/docs/iOS/client/state-layer/state-layer-overview.md

```diff
@@ -0,0 +1,229 @@
+
+Every async method is safe to be called from any thread. Observable state objects are isolated to the main actor. This gives a guarantee that observable state changes and reads happen on the main actor. While state objects are isolated to the main actor, we can still access the state from non-main actor contexts which just requires to use the `await` keyword.
+
+Let's consider an example below where we ask to load messages. When the `loadMessages(with:)` returns, we are guaranteed that the observable state object has all the loaded messages available.
+
+```swift
+try await chat.loadMessages(with: MessagesPagination(pageSize: pageSize))
+let allMessages = chat.state.messages
+```
+
+## Types Representing the State
+
+The state layer has objects representing different use-cases which enables accessing and manipulating data the Stream API provides. These types follow an architecture where high level types have methods for interacting with the Stream API and the state counterpart keeps the currently loaded data. The state object is observable and is backed by the local persistent store.
+
+* ChannelList
+  * ChannelListState
+* Chat
+  * ChatState
+  * MessageState
+* ConnectedUser
+  * ConnectedUserState
+* MemberList
+  * MemberListState
+* MessageSearch
+  * MessageSearchState
+* ReactionList
+  * ReactionListState
+* UserList
+  * UserListState
+* UserSearch
+  * UserSearchState
+
+These objects are created through factory methods in `ChatClient`. Factory methods create instances of `ChannelList`, `Chat`, `ConnectedUser`, `MemberList`, `MessageSearch`, `ReactionList`, `UserList`,  and `UserSearch`. The observable state is accessed through `state` properties what return the respective state object. `ChannelList`, `Chat`, `MemberList`, `ReactionList`, and `UserList` have a `get()` method for fetching the default set of data from the Stream API.
+
+### ChannelList
+
+`ChannelList` and its `ChannelListState` enable querying a list of channels from the Stream API and provide an interface for loading channels in a paginated manner.
+
+```swift
+let query = ChannelListQuery(
+    filter: .and([
+        .equal(.type, to: .messaging), 
+        .containMembers(userIds: ["thierry"])
+    ]),
+    sort: [Sorting(key: .lastMessageAt, isAscending: false)],
+    pageSize: 10
+)
+let channelList = chatClient.makeChannelList(with: query)
+// local state of the channel list (state from previous sessions)
+var channels = channelList.state.channels
+// fetch the first page of channels from the Stream API
+try await channelList.get()
+// fetch more channels
+try await channelList.loadMoreChannels()
```

Comment:
Random thought: Now that I look into these 2 together, probably `getMoreChannels()` makes more sense 😅 Since get and load are pretty much the same thing, the naming could be more consistent across the different APIs 🤔